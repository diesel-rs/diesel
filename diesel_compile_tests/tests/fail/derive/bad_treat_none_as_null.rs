#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
        name -> Text,
    }
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
#[diesel(treat_none_as_null("true"))]
//~^ ERROR: expected `=`
struct UserForm1 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
#[diesel(treat_none_as_null)]
//~^ ERROR: unexpected end of input, expected `=`
struct UserForm2 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
#[diesel(treat_none_as_null = "foo")]
//~^ ERROR: expected boolean literal
struct UserForm3 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[diesel(table_name = users)]
struct UserForm4 {
    id: i32,
    #[diesel(treat_none_as_null = true)]
    name: String,
    //~^ ERROR: expected `treat_none_as_null` field to be of type `Option<_>`
}

fn main() {}
