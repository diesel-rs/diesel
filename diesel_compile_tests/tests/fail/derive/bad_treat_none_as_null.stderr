error: expected `=`
  --> tests/fail/derive/bad_treat_none_as_null.rs:13:28
   |
13 | #[diesel(treat_none_as_null("true"))]
   |                            ^

error: unexpected end of input, expected `=`
       help: The correct format looks like `#[diesel(treat_none_as_null = true)]`
  --> tests/fail/derive/bad_treat_none_as_null.rs:21:28
   |
21 | #[diesel(treat_none_as_null)]
   |                            ^

error: expected boolean literal
  --> tests/fail/derive/bad_treat_none_as_null.rs:29:31
   |
29 | #[diesel(treat_none_as_null = "foo")]
   |                               ^^^^^
